### PR TITLE
[x-plane] Update to 3.0.3

### PR DESCRIPTION
**Pull request by siavee**
**Describe the pull request**

- What does your PR fix?

  Updated x-plane package to latest (3.0.3) & changed macOS lib deployment to include .framework directories as it is in the original package.

- Which triplets are supported/not supported? Have you updated the CI baseline?

  Package is header & precompiled lib only

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  
  Yes

### CODE REVIEW

File: ports/x-plane/portfile.cmake [hunk 1]
```diff
@@ -1,37 +1,39 @@
-if (VCPKG_TARGET_ARCHITECTURE STREQUAL x86)
-    message(FATAL_ERROR "the x-plane SDK cannot be built for the x86 architecture")
-endif()
+vcpkg_fail_port_install(MESSAGE "The X-Plane SDK cannot be built for the x86 architecture" ON_ARCH "x86")
```
Comment (NancyLi1013) [inline ports/x-plane/portfile.cmake:1]:
```suggestion
vcpkg_fail_port_install(ON_ARCH "x86")
```

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/x-plane/portfile.cmake (hunk 1 above)
Comment (siavee) [reply ports/x-plane/portfile.cmake:1]: Thanks for the review, so we don't need to show any message to the user?

**Comment by NancyLi1013** [comment]
Hi @siavee 

Thanks for your PR.

Could you please help check the build status on arm, uwp and linux for the new version?

If they have passed, please remove these codes from ci.baseline.txt.

```
x-plane:arm64-windows=fail
x-plane:arm-uwp=fail
x-plane:x64-linux=fail
x-plane:x86-windows=fail
```
`x-plane:x86-windows=fail` can be removed now since you have added Supports: !x86 in vcpkg.json.

**Comment by siavee** [comment]
Hi @NancyLi1013 thanks for the review, the SDK is x64 only so arm is definitely unsupported. I've tested x64-linux to be working find. I'll update the requested files in a bit.

File: ports/x-plane/portfile.cmake (hunk 1 above)
Comment (siavee) [reply ports/x-plane/portfile.cmake]: I've updated the portfile, the MESSAGE directive is removed and added unsupported architectures.

File: ports/x-plane/vcpkg.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "name": "x-plane",
+  "version-string": "3.0.3",
+  "description": "The X-Plane Plugin SDK",
+  "homepage": "https://developer.x-plane.com/sdk/",
+  "supports": "!x86"
```
Comment (NancyLi1013) [inline ports/x-plane/vcpkg.json:6]:
```suggestion
  "supports": "!(x86|arm|wasm32)"
```

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/x-plane/portfile.cmake [hunk 3]
```diff
@@ -1,37 +1,39 @@
...
-    file(COPY ${SOURCE_PATH}/Libraries/Mac/XPWidgets.framework/ DESTINATION ${CURRENT_PACKAGES_DIR}/lib/)
-    file(COPY ${SOURCE_PATH}/Libraries/Mac/XPLM.framework/ DESTINATION ${CURRENT_PACKAGES_DIR}/debug/lib/)
-    file(COPY ${SOURCE_PATH}/Libraries/Mac/XPWidgets.framework/ DESTINATION ${CURRENT_PACKAGES_DIR}/debug/lib/)
+elseif (VCPKG_TARGET_IS_OSX)
+    file(COPY ${SOURCE_PATH}/Libraries/Mac/ DESTINATION ${CURRENT_PACKAGES_DIR}/lib/)
+    file(COPY ${SOURCE_PATH}/Libraries/Mac/ DESTINATION ${CURRENT_PACKAGES_DIR}/debug/lib/)
 endif()
 
+# Handle copyright
 file(INSTALL ${SOURCE_PATH}/license.txt DESTINATION ${CURRENT_PACKAGES_DIR}/share/x-plane/ RENAME copyright)
```
Comment (NancyLi1013) [inline ports/x-plane/portfile.cmake:39]:
```suggestion
file(INSTALL ${SOURCE_PATH}/license.txt DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
```

**Comment by NancyLi1013** [comment]
Please help update the test result on Linux, thanks.

**Comment by siavee** [comment]
Roger, in a bit

**Comment by siavee** [comment]
Working fine in Linux.

File: ports/x-plane/portfile.cmake [hunk 4]
```diff
@@ -1,37 +1,39 @@
...
     file(COPY ${SOURCE_PATH}/Libraries/Win/XPWidgets_64.lib DESTINATION ${CURRENT_PACKAGES_DIR}/debug/lib/)
-elseif (VCPKG_CMAKE_SYSTEM_NAME STREQUAL "Darwin")
-    file(COPY ${SOURCE_PATH}/Libraries/Mac/XPLM.framework/ DESTINATION ${CURRENT_PACKAGES_DIR}/lib/)
-    file(COPY ${SOURCE_PATH}/Libraries/Mac/XPWidgets.framework/ DESTINATION ${CURRENT_PACKAGES_DIR}/lib/)
-    file(COPY ${SOURCE_PATH}/Libraries/Mac/XPLM.framework/ DESTINATION ${CURRENT_PACKAGES_DIR}/debug/lib/)
-    file(COPY ${SOURCE_PATH}/Libraries/Mac/XPWidgets.framework/ DESTINATION ${CURRENT_PACKAGES_DIR}/debug/lib/)
+elseif (VCPKG_TARGET_IS_OSX)
+    file(COPY ${SOURCE_PATH}/Libraries/Mac/ DESTINATION ${CURRENT_PACKAGES_DIR}/lib/)
+    file(COPY ${SOURCE_PATH}/Libraries/Mac/ DESTINATION ${CURRENT_PACKAGES_DIR}/debug/lib/)
 endif()
```
Comment (NancyLi1013) [inline ports/x-plane/portfile.cmake:36]: It seems like there is no Linux, only Windows and OSX. Are you missing something here?

Comment (siavee) [reply ports/x-plane/portfile.cmake:36]: On Linux, there is nothing to copy, to quote from the SDK docs:

> Note: there are no import/link-time libraries for Linux; on Linux, plugins simply leave SDK symbols undefined and they are discovered at runtime.

Comment (siavee) [reply ports/x-plane/portfile.cmake:36]: The docs that I refer to is the README.txt file inside the archive.

**Review by NancyLi1013 — APPROVED** [review]

**Comment by NancyLi1013** [comment]
LGTM now, thanks for your PR @siavee.

**Comment by siavee** [comment]
Thanks for the review too @NancyLi1013 :)

**Comment by BillyONeal** [comment]
Thank you!

**Comment by siavee** [comment]
And thank you to you too!

**Pull request merged**